test/apps/iodemo/io_demo.cc

```diff
@@ -804,45 +804,43 @@ class P2pDemoCommon : public UcxContext {
         ~ValidateError() throw() {
         }
 
+        virtual const char *what() const throw() {
+            return _str.c_str();
+        }
+
     protected:
-        std::stringstream _ss;
+        std::string _str;
```

Comment:
minor better to keep as it was, i.e. `std::stringstream` to avoid extra copy

Reply:
it was a bug, `std::stringstream::str()` returns `std::string` and anyway it has extra copy
```
string str() const;
```

Reply:
> anyway it has extra copy

actually not nessesary, see return value optimization (RVO), but I'm not sure if this is the case

Reply:
yes, RVO should help here.
but the previous code was incorrect it returns a pointer to the inner object (`const char *` from `c_str()``) from the function `what()`.